agent/api/registryauth.go

```diff
@@ -24,6 +24,7 @@ import (
 type RegistryAuthenticationData struct {
 	Type        string       `json:"type"`
 	ECRAuthData *ECRAuthData `json:"ecrAuthData"`
+	ASMAuthData *ASMAuthData `json:"asmAuthData"`
```

Comment:
Will this be unmarshalled from the payload message, can we modify the test to cover this?

Reply:
yes it will and completely missed this. will add test.